corehq/apps/reports/v2/filters/xpath_column.py

```diff
@@ -0,0 +1,142 @@
+
+    @classmethod
+    def get_context(cls):
+        return {
+            'title': cls.title,
+            'name': cls.name,
+            'choices': [
+                {
+                    'title': c.title,
+                    'name': c.name
+                } for c in cls.choices
+            ],
+        }
+
+
+class TextXpathColumnFilter(BaseXpathColumnFilter):
+    title = ugettext_lazy("Text")
+    name = 'xpath_column_text'
+    data_type = DataType.TEXT
+    choices = [
+        ChoiceMeta(ugettext_lazy("Equals"), 'text_equals', '='),
+        ChoiceMeta(ugettext_lazy("Does not equal"), 'text_not_equals', '!='),
+    ]
+
+    @classmethod
+    def format_value(cls, value):
+        return "'{}'".format(value)
+
+
+class NumericXpathColumnFilter(BaseXpathColumnFilter):
+    title = ugettext_lazy("Number")
+    name = 'xpath_column_number'
+    data_type = DataType.NUMERIC
+    choices = [
+        ChoiceMeta(ugettext_lazy("Is equal to"), 'num_equals', '='),
+        ChoiceMeta(ugettext_lazy("Is less than"), 'num_less_than', '<'),
+        ChoiceMeta(ugettext_lazy("Is greater than"), 'num_greater_than', '>'),
+        ChoiceMeta(ugettext_lazy("Is not equal to"), 'num_not_equals', '!='),
+    ]
+
+    @staticmethod
+    def _get_number(value):
+        try:
+            return int(value)
+        except ValueError:
+            return float(value)
+
+    @classmethod
+    def format_value(cls, value):
+        return "'{}'".format(value) if value == '' else cls._get_number(value)
+
+
+class ColumnXpathExpressionBuilder(object):
+
+    def __init__(self, column_context, column_filters):
+        self.column_name = column_context['name']
+        self.clause = column_context.get('clause', Clause.ALL)
+        self.column_filters = column_filters
+        self.applied_filters = []
+        for filter_context in column_context.get('filters', []):
+            self.applied_filters.append(AppliedFilterContext(
+                filter_name=filter_context['filterName'],
+                choice_name=filter_context['choiceName'],
+                value=filter_context['value'],
+            ))
+
+    def _get_filter_by_name(self, filter_name):
+        name_to_class = {f.name: f for f in self.column_filters}
+        try:
+            column_filter = name_to_class[filter_name]
+            return column_filter
+        except (KeyError, NameError):
+            raise ColumnFilterNotFound(
+                "Could not find the column filter '{}'.".format(filter_name)
+            )
+
+    def get_expression(self):
+        parts = []
+        for context in self.applied_filters:
+            column_filter = self._get_filter_by_name(context.filter_name)
+            try:
+                parts.append(column_filter.get_expression(
+                    self.column_name,
+                    context.choice_name,
+                    context.value
+                ))
+            except ValueError:
+                continue  # fail silently
```

Comment:
Is this catching a `ValueError` that would result from `float('foo')` in the `NumericXpathColumnFilter`? If so, it might make more sense to catch this above so that it is obvious what inputs you are explicitly ignoring.

I'm also curious why you would make this fail silently instead of telling the user that they can't do what they are trying to do.

Reply:
see the above comment. Also I'm adding typing to the UI that will enforce numeric values in an update to this PR 😄